app/src/main/java/com/epfl/beatlink/ui/authentication/SignUp.kt

```diff
@@ -37,20 +37,23 @@ import androidx.compose.ui.text.input.PasswordVisualTransformation
 import androidx.compose.ui.text.withStyle
 import androidx.compose.ui.unit.dp
 import androidx.lifecycle.viewmodel.compose.viewModel
+import com.epfl.beatlink.model.profile.ProfileData
 import com.epfl.beatlink.ui.components.CustomInputField
 import com.epfl.beatlink.ui.navigation.NavigationActions
 import com.epfl.beatlink.ui.navigation.Screen
 import com.epfl.beatlink.ui.spotify.SpotifyAuth
 import com.epfl.beatlink.ui.theme.PrimaryGradientBrush
 import com.epfl.beatlink.viewmodel.auth.FirebaseAuthViewModel
+import com.epfl.beatlink.viewmodel.profile.ProfileViewModel
 import com.epfl.beatlink.viewmodel.spotify.auth.SpotifyAuthViewModel
 
 @Composable
 fun SignUpScreen(
     navigationActions: NavigationActions,
     spotifyAuthViewModel: SpotifyAuthViewModel,
+    profileViewModel: ProfileViewModel,
     firebaseAuthViewModel: FirebaseAuthViewModel =
-        viewModel(factory = FirebaseAuthViewModel.Factory)
+        viewModel(factory = FirebaseAuthViewModel.Factory),
```

Comment:
I think this can be done in the BeatLinkApp.kt file at the top (see ProfileViewModel and PlaylistViewModel)

Reply:
This approach works, but I will adjust it to align with how the majority have initialized the viewmodel.